pkg/controllers/configobservation/console/observe_consoleurl_test.go

```diff
@@ -17,38 +17,51 @@ import (
 
 func TestObserveConsoleURL(t *testing.T) {
 	existingConfig := configWithConsoleURL("https://teh.console.my")
+	noConfig := map[string]interface{}(nil)
 
 	tests := []struct {
 		name                string
 		consoleConfig       *configv1.ConsoleStatus
+		clusterVersion      *configv1.ClusterVersionStatus
 		existingConfig      map[string]interface{}
 		expectedConfig      map[string]interface{}
 		expectedErrs        []string
 		expectedUpdateEvent bool
 	}{
 		{
-			name:           "NoConsoleConfig",
+			name:           "NoConsoleConfigConsoleCapabilityEnabled",
```

Comment:
what happens when the capability is not enabled but someone creates the CRD and the object?

Reply:
Then I think the cluster would be in an inconsistent state since the the CRD would be there but quite a lot of other resrouces would be missing. 
Also is that even a valid scenario? meaning that only a cluster admin can create CRDs, not mentioning that he would/should be the one that decides which set of capabilities should be enabled when provisioning the cluster.

Reply:
> Also is that even a valid scenario?

Unless you can prevent it, yes.

Add a unit test, please.

> cluster admin can create CRDs

Cluster admin and likely anyone installing an operator.